Makefile

```diff
@@ -0,0 +1,57 @@
+PROJECT := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+
+SRC := $(PROJECT)src
+TESTS := $(PROJECT)tests
+ALL := $(SRC) $(TESTS)
+
+export PYTHONPATH = $(PROJECT):$(PROJECT)/lib:$(SRC)
+
+update-dependencies:
+	uv lock -U --no-cache
+
+generate-requirements:
+	uv pip compile -q --no-cache pyproject.toml -o requirements.txt
+	uv pip compile -q --no-cache --all-extras pyproject.toml -o requirements-dev.txt
+
+lint:
+	uv tool run ruff check $(ALL)
+	uv tool run ruff format --check --diff $(ALL)
+
+fmt:
+	uv tool run ruff check --fix-only $(ALL)
+	uv tool run ruff format $(ALL)
+
+unit:
+	uv run --all-extras \
+		coverage run \
+		--source=$(SRC) \
+		-m pytest \
+		--ignore=$(TESTS)/integration \
+		--tb native \
+		-v \
+		-s \
+		$(ARGS)
+	uv run --all-extras coverage report
+
+integration:
+	uv run --all-extras \
```

Comment:
Jon and I disagree on this, but I think [this should be `uv run --frozen ...`](https://github.com/canonical/parca-k8s-operator/pull/339#discussion_r1757214864).  To me, updating dependencies should be an explicit opt-in rather than a side effect.  The main case for this is debugging an existing charm - if I pull an old charm release and start reproducing an issue, I want `make integration` to use exactly what was released.

To be fair though, `uv run` [often does what a dev would want](https://github.com/canonical/parca-k8s-operator/pull/339#discussion_r1757302817)